src/components/NavBar/NavBar.jsx

```diff
@@ -60,6 +61,13 @@ const NavBar = () => {
     setShowConfirmationModal(false);
   };
 
+  const { profileData } = useContext(SignedInUserContext);
+  const [profileImg, setProfileImg] = useState(localStorage.getItem('profileImage'));
```

Comment:
You already have this information in `profileData`. You don't need to store it in `localStorage` as well.